.github/workflows/ci.yml

```diff
@@ -14,7 +14,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - name: Check out code
-      uses: actions/checkout@v2
+      uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
```

Comment:
As mentioned, I understand the advantage of pinning dependencies to specific versions, but I think it's too much work to maintain. I think I would prefer just pinning to a major version.

Reply:
This is done for security purposes. The contents of an action in a specific version can change, but not when you pin it to a commit SHA. Maintenance is required either way in any cadence you prefer.

If you are worried about that, I can setup dependabot to check for actions upgrades and assign myself as assignee.

Reply:
Very good point.

Reply:
it's not a big deal, just a matter of preference. so we can try it out for now. I'll mark this as resolved.